editor/effects.js

```diff
@@ -141,10 +141,14 @@ export default {
 	},
 	TRASH_POST_SUCCESS( action ) {
 		const { postId, postType } = action;
-		window.location.href = getWPAdminURL( 'edit.php', {
-			trashed: 1,
-			post_type: postType,
-			ids: postId,
+
+		// Delay redirect to ensure store has been updated with the successful trash.
```

Comment:
One thing that might be nice to have with the effects is the `next` function provided to middlewares, so that we could perform effects before _or after_ the original action has been applied.